ligo/src/management.mligo

```diff
@@ -4,36 +4,74 @@
 // Corresponds to Management.hs module
 
 #include "types.mligo"
+#include "common.mligo"
 
-
-// -----------------------------------------------------------------
-// Shared Function
-// -----------------------------------------------------------------
 #if !MANAGEMENT_H
 #define MANAGEMENT_H
 
-[@inline]
-let ensure_not_migrated (storage : storage): storage =
-  match storage.migration_status with
-      Not_in_migration -> storage
-    | MigratingTo (p) -> storage
-    | MigratedTo (p) -> (failwith ("MIGRATED") : storage)
-
-#endif  // MANAGEMENT_H included
-
-
+(*
+ * Auth checks for admin and store the address in parameter to the
+ * 'pending_owner' field in storage.
+ *)
 let transfer_ownership
     (param, store : transfer_ownership_param * storage) : return =
-  not_implemented("transfer_ownership")
+      if authorize_admin(store)
+        then
+          (([] : operation list), { store with pending_owner = param ; })
+        else (failwith("NOT_ADMIN") : return)
 
+(*
+ * Auth check for pending admin and copies the value in 'pending_owner' field
+ * to 'admin' field. The 'pending_owner' field is left with the address of the
+ * new admin.
+ *)
 let accept_ownership(param, store : unit * storage) : return =
-  not_implemented("transfer_ownership")
+  if store.pending_owner = Tezos.sender
+    then (([] : operation list), { store with admin = store.pending_owner })
+    else (failwith("NOT_PENDING_ADMIN") : return)
 
+(*
+ * Auth check for admin and sets the migration status to 'MigratingTo' using
+ * the address from parameter, overwriting any address from any previous
+ * 'Migrate' calls.
+ *)
 let migrate(param, store : migrate_param * storage) : return =
-  not_implemented("migrate")
+  if authorize_admin(store)
+    then (([] : operation list), { store with migration_status = MigratingTo (param) })
+    else (failwith("NOT_ADMIN") : return)
 
+(*
+ * Auth check for new contract and sets the migration_status to 'Migrated', using
+ * the address from the earlier value in the migration_status field.
+ *)
 let confirm_migration(param, store : unit * storage) : return =
-  not_implemented("confirm_migration")
+  match store.migration_status with
+    Not_in_migration -> (failwith("NOT_MIGRATING") : return)
+  | MigratingTo (new_addr) -> if new_addr = Tezos.sender
+      then (([] : operation list), { store with migration_status = MigratedTo (new_addr) })
+      else (failwith("NOT_MIGRATION_TARGET") : return)
+  | MigratedTo (new_addr)  -> (failwith("MIGRATED") : return)
```

Comment:
I guess this clause will never happen now that `ensure_not_migrated` is executed before this, but I suppose that's fine.